stock_reserve_rule/models/stock_reserve_rule.py

```diff
@@ -333,3 +398,69 @@ def _apply_strategy_full_bin(self, quants):
+        limit = self.tolerance_requested_limit
+        computation = self.tolerance_requested_computation
+        if limit == "no_tolerance" or float_compare(tolerance, 0, rounding) == 0:
+            return float_compare(need, product_qty, rounding) == 0
+        elif limit == "upper_limit":
+            if computation == "percentage":
+                # need + rounding < product_qty <= need * (100 + tolerance) / 100
+                return (
+                    float_compare(need, product_qty, rounding) == -1
+                    and float_compare(
+                        product_qty, need * (100 + tolerance) / 100, rounding
+                    )
+                    <= 0
+                )
+            else:
+                # need + rounding < product_qty <= need + tolerance
+                return (
+                    float_compare(need, product_qty, rounding) == -1
+                    and float_compare(product_qty, need + tolerance, rounding) <= 0
+                )
+        elif limit == "lower_limit":
+            if computation == "percentage":
+                # need * (100 - tolerance) / 100 <= product_qty < need - rounding
+                return (
+                    float_compare(need * (100 - tolerance) / 100, product_qty, rounding)
+                    <= 0
+                    and float_compare(product_qty, need, rounding) == -1
+                )
+            # computation == "absolute"
+            else:
+                # need - tolerance <= product_qty < need - rounding
+                return (
+                    float_compare(need - tolerance, product_qty, rounding) <= 0
+                    and float_compare(product_qty, need, rounding) == -1
+                )
+
+    def _apply_strategy_full_lot(self, quants):
+        need = yield
+        # We take goods only if we empty the bin.
+        # The original ordering (fefo, fifo, ...) must be kept.
+        product = fields.first(quants).product_id
+        rounding = product.uom_id.rounding
+        if product.tracking == "lot":
+            for lot, lot_quants in quants.filtered(
+                lambda quant, product_id=product.id: quant.product_id.id == product_id
+            ).group_by_lot():
+                for lot_quant in lot_quants:
+                    product_qty = sum(lot_quant.mapped("quantity"))
+                    lot_quantity = sum(lot_quant.mapped("quantity")) - sum(
+                        lot_quant.mapped("reserved_quantity")
+                    )
+                    if (
+                        lot
+                        and self._compare_with_tolerance(need, product_qty, rounding)
+                        and lot_quantity > 0
+                    ):
+                        need = (
+                            yield lot_quant.location_id,
+                            lot_quantity,
+                            need,
+                            lot,
+                            None,
+                        )  # noqa
```

Comment:
```suggestion
                product_qty = sum(lot_quants.mapped("quantity"))
                if not lot or not self._compare_with_tolerance(need, product_qty, rounding):
                    continue
                for lot_quant in lot_quants:
                    lot_quantity = lot_quant.quantity - lot_quant.reserved_quantity
                    need = yield (
                            lot_quant.location_id,
                            lot_quantity,
                            need,
                            lot,
                            None,
                        )  # noqa
```

Reply:
@jbaudoux Could you check my proposed version of the function?

Reply:
@jbaudoux Your code has an error.

Reply:
@geomer198 was this addressed?